### PR TITLE
Bugfix and improvements in visualize package

### PR DESCRIPTION
**Pull request by nontas**
This PR fixes the following bugs:
* Corrects the viewing behaviour of `PointCloud` and `PointGraph` objects in case `image_view=True`, i.e. the way that the y axis gets inverted.
* Prints the `LandmarkGroup` name in the legend (before, it was only printing 'group').

It also adds the following:
*  Adds `axes_x_ticks` and  `axes_y_ticks` to all `view()` methods. This is especially useful when plotting curves (e.g. CED).
* Adds a new `plot_curve()` method in `menpo.visualize` that can plot multiple curves.
* Changes the `axes_x_limits` and `axes_y_limits` behaviour. Specifically, if `float`, then it sets padding around the object that is visualized defined as a percentage of the object's width. If `tuple` or `list`, then they define the axes limits. If ``None``, then the limits are set automatically.
* Adds the ability to print numbers on the points of a `PointCloud` or a `PointGraph`.

Finally, note that this PR also changes the `matplotlib` dependency to:
```
matplotlib>=1.5,<1.6
```
which removes an annoying warning that they used to print.

### CODE REVIEW

**Comment by patricksnape** [comment]
Right, lots of changes here - will check this out later tonight.

File: setup.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
     include_dirs = [np.get_include()]
     install_requires = ['numpy>=1.10,<1.11',
                         'scipy>=0.16,<0.17',
-                        'matplotlib>=1.4,<1.5',
+                        'matplotlib>=1.5,<1.6',
```
Comment (jabooth) [inline setup.py]: we need this in `conda/meta.yaml` too

**Comment by patricksnape** [comment]
Right well this passed so I'm going to +1 this @jabooth for the 0.6.1 release

**Comment by jabooth** [comment]
Sounds good, thanks @nontas and @patricksnape +1

**Pull request merged**